### PR TITLE
feat(core): view engine - support content projection

### PR DESCRIPTION
**Pull request by tbosch**
First commit is from https://github.com/angular/angular/pull/14197

### CODE REVIEW

File: modules/@angular/core/src/view/types.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export interface NodeDef {
   reverseChildIndex: number;
   flags: NodeFlags;
   parent: number;
+  ngContentIndex: number;
```
Comment (tbosch) [inline modules/@angular/core/src/view/types.ts]: add comment

File: modules/@angular/core/src/view/types.ts [hunk 2]
```diff
@@ -229,6 +232,8 @@ export enum QueryBindingType {
   All
 }
 
+export interface NgContentDef { index: number; }
```
Comment (tbosch) [inline modules/@angular/core/src/view/types.ts]: add comment about the index vs ngContentIndex

**Review by vicb — APPROVED** [review]

File: modules/@angular/core/src/view/types.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/src/view/types.ts]: Done

File: modules/@angular/core/src/view/types.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/view/types.ts]: Done

**Comment by DzmitryShylovich** [comment]
@tbosch https://github.com/angular/angular/issues/4946#issuecomment-276243091 what do you think?

**Comment by tbosch** [comment]
@DzmitryShylovich Added a comment. I don't think we should do this now as it is a bit complicated. Also talked to @mhevery about it.

**Comment by tbosch** [comment]
@mhevery This marker code-review/pullapprove wrong (@vicb approved it) , @IgorMinar will look into it...

**Pull request closed without merging**